common/src/java/com/zimbra/common/util/ZimbraCookie.java

```diff
@@ -76,7 +76,12 @@ private static void addCookie(HttpServletResponse response, String name, String
         }
         ZimbraCookie.setAuthTokenCookieDomainPath(cookie, ZimbraCookie.PATH_ROOT);
 
-        cookie.setSecure(secure);
+        String cookieVal = LC.zimbra_same_site_cookie.value();
+        if (!StringUtil.isNullOrEmpty(cookieVal)) {
+            cookie.setSecure(true);
+        } else {
+            cookie.setSecure(secure);
+        }
```

Comment:
If we are going to maintain this logic here, we should ensure that all of the code is actually using this method to add cookies. I was able to find at least one example where `setAuthTokenCookieDomainPath` is called directly and not followed by a call to `addCookie`. (see https://github.com/Zimbra/zm-saml-consumer-store/blob/b37b0409787a490e5824b78c2e6807c1320313fa/src/java/com/zimbra/cs/security/saml/SamlLogoutReceiverHandler.java#L95)

Reply:
@jeastman  moved implementation in addCookie method also every module mentor have to backport these changes to support SameSite=Strict.